### ISSUE TITLE
Expose Native Buffers ID for OpenGL-CUDA Interoperability 

### ISSUE THREAD

**Issue by QuanyiLi** [panda3d/panda3d#1468]
## Description
As discussed in https://discourse.panda3d.org/t/allowing-torch-tensorflow-to-directly-access-rendered-image-in-gpu/29119/7, it is possible to do interoperation between OpenGL and CUDA with native buffer ID. However, the related Python API lacks in Panda. Could you please provide such an interface?

## Use Case

I am testing OpenGL and CUDA interoperability with pure Python interfaces provided by Panda3D and CuPY, respectively. Such a feature allows nowadays deep learning frameworks, like Tensorflow/PyTorch to manipulate the rendered content on GPU, which could improve the efficiency of visual data collection and model training by a large margin.

**Comment by rdb** [comment]
We just need to add a virtual get_native_id method to BufferContext, similar to how we have it in TextureContext, and implement it in the various buffer-related GL subclasses.

This could then easily be supported for vertex buffers, index buffers, and SSBOs.

**Comment by Spartan-X1** [comment]
Hi @rdb, I'm interested in working on this issue. I'm an absolute beginner to open source, but I have experience with in C++ & Python and am eager to learn more about graphics programming. Since this issue is from 2023, I wanted to check if it's still available for contribution. I'd appreciate any guidance on getting started, particularly about the BufferContext implementation you mentioned.

**Comment by rdb** [comment]
Yes, feel free to work on this. You can look at the existing get_native_id code and implement it for the buffer-related classes.

**Comment by Spartan-X1** [comment]
Thank you for the opportunity, I'll study the existing get_native_id implementation in TextureContext and implement it for the buffer-related GL subclasses as suggested.

**Comment by Spartan-X1** [comment]
Hello @rdb , I am sorry to bother you again but I just need to expose the OpenGL native ID in just these files right :
glIndexBufferContext_src ,  glBufferContext_src & glVertexBufferContext_src  & a base file code for BufferContext ?

**Comment by rdb** [comment]
Yup, that sounds about right!

**Comment by Spartan-X1** [comment]
I made a PR , please review it & if something is wrong please tell me

**Comment by rdb** [comment]
I pushed support for `get_native_id` for all relevant resources (sampler objects, IBOs, VBOs, SSBOs)
